acceptance-tests/src/test/java/io/blueocean/ath/offline/BitbucketServerTest.java

```diff
@@ -0,0 +1,66 @@
+package io.blueocean.ath.offline;
+
+import com.cdancy.bitbucket.rest.BitbucketClient;
+import com.cdancy.bitbucket.rest.options.CreateRepository;
+import com.google.inject.Inject;
+import io.blueocean.ath.ATHJUnitRunner;
+import io.blueocean.ath.Login;
+import io.blueocean.ath.WaitUtil;
+import io.blueocean.ath.pages.blue.DashboardPage;
+import org.apache.log4j.Logger;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.By;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.io.IOException;
+
+@Login
+@RunWith(ATHJUnitRunner.class)
+public class BitbucketServerTest {
+
+    private static Logger LOGGER = Logger.getLogger(BitbucketServerTest.class);
+
+    @Inject
+    WaitUtil wait;
+
+    @Inject
+    DashboardPage dashboardPage;
+
+    @Test
+    public void testJenkinsfileCreate() {
+        BitbucketClient client = BitbucketClient.builder()
+            .endPoint("http://127.0.0.1:7990/bitbucket")
+            .credentials("admin:admin").build();
+
+        String project = "PROJECT_1";
+        String repo = "testJenkinsfileCreate";
+        client.api().repositoryApi().delete(project, repo);
+        client.api().repositoryApi().create(project, CreateRepository.create(repo, true));
+
+        dashboardPage.open();
+        wait.until(ExpectedConditions.visibilityOf(dashboardPage.newPipelineButton))
+            .click();;
+        LOGGER.info("Clicked on new pipeline button");
+
+        wait.until(By.xpath("//span[text()='Bitbucket Server']")).click();
```

Comment:
Any way of not using text for this? I know it _should_ be fine, using the english translation, but still

Reply:
I agree we should stop selecting on text values. Please :) They are brittle, as we were originally using "Github" in our UI and then changed it once someone pointed out that "GitHub" is the proper branding. I don't see any reason why we shouldn't be making using liberal use of semantic CSS class names in our UI code (e.g. "bitbucket-server-scm-button") and selecting on that. If they are missing me, we should feel free to add them. Also, clicking on a `span` can result in "element not clickable" errors, we should really click on `button` or anchor elements IMO.